### ISSUE TITLE
False positive: FallThrough for switch block with "switch rules"

### ISSUE THREAD

**Issue by anthonyvdotbe** [google/error-prone#1649]
### Description of the problem / feature request:

A false positive for `FallThrough` is reported for switch blocks with so-called switch rules.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

````java
class Foo {

    void bar(int n) {
        switch (n) {
            case 42 -> System.out.println("baz");
            default -> throw new IllegalArgumentException();
        }
    }
}
````

### What version of Error Prone are you using?

2.4.0

###  Have you found anything relevant by searching the web?

No.

**Comment by cushon** [comment]
Fixed in b84b80c2624219894ea67ee6beee97d1ec01860b
